pkg/util/util.go

```diff
@@ -314,11 +314,16 @@ func GetDNS1123Name(str string) string {
 	withReplacedChars := strings.Replace(
 		nonAllowedCharsRegex.ReplaceAllString(str, "-"),
 		"--", "-", -1)
-	return removeNonAlphaSuffix(removeNonAlphaPrefix(withReplacedChars))
+	name := strings.ToLower(removeNonAlphaSuffix(removeNonAlphaPrefix(withReplacedChars)))
+	// if the name is all numeric
+	if len(str) != 0 && len(name) == 0 {
```

Comment:
How does checking the length of variables help us determine if the name is all numeric? What am I missing here?

Reply:
If the len of the original string is not 0 is the control condition just or make sure no one passed a blank directory,
Then after converting it to DNS1123 - if all are numeric the “removeNonAlphaprefix” would remove all characters as everything is non alpha hence returning a blank string.